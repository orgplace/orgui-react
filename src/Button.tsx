import classNames from 'classnames';
import React from 'react';
import { OuiColorVariationAttributes } from '.';

const OuiButton: React.FC<React.ButtonHTMLAttributes<HTMLButtonElement> & OuiColorVariationAttributes> = (props) => {
  const {colorVariation, ...buttonProps} = props;
  return (
    <button {...buttonProps} className={classNames(
      'ouiButton',
      colorVariation,
      buttonProps.className,
    )}>{buttonProps.children}</button>
  );
}

const OuiLinkButton: React.FC<React.ButtonHTMLAttributes<HTMLAnchorElement> & OuiColorVariationAttributes> = (props) => {
  const {colorVariation, ...anchorProps} = props;
  return (
    <a {...anchorProps} className={classNames(
      'ouiButton',
      colorVariation,
      anchorProps.className,
    )}>{anchorProps.children}</a>
  );
}

export { OuiButton, OuiLinkButton };
