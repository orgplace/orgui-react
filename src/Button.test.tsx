import React from 'react'
import { OuiButton } from './Button'
import renderer from 'react-test-renderer';

test('OuiButton', function() {
	const component = renderer.create(<OuiButton>test</OuiButton>);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
