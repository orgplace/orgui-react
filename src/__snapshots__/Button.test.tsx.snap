// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiButton 1`] = `
<button
  className="ouiButton"
>
  test
</button>
`;
